numpy/core/function_base.py

```diff
@@ -123,25 +123,47 @@ def linspace(start, stop, num=50, endpoint=True, retstep=False, dtype=None,
     if num < 0:
         raise ValueError("Number of samples, %s, must be non-negative." % num)
     div = (num - 1) if endpoint else num
+    # Make sure one can use variables that have an __array_interface__, gh-6634
+    start = asanyarray(start)
+    stop  = asanyarray(stop)
+
+    # check if dealing with datetime64 or timedelta64
+    is_datelike = start.dtype.kind in 'Mm'
+    if is_datelike:
+        # For datetime/timedelta, dtype must be specified
+        # start and stop will first be cast to this dtype.
+        # Warning: dtype must have high enough precision so
+        # that it is at all possible to draw num equispaced
+        # samples.
+        assert dtype is not None, "linspace with {} objects cannot be called "\
+            "with dtype=None".format(start.dtype)
+        start = start.astype(dtype)
+        stop = stop.astype(dtype)
+        # y cannot be cast to datetime64/timedelta64, but left
+        # as int64, since later on we will do y * delta, and
```

Comment:
(just for you, technically, on 32bit platforms, arange probably returns int32. It does not matter though.)

Reply:
Debsankha, do you know right away why making `y` a datetime/timedelta here fails later?

Reply:
@seberg : Sorry for the late reply. Making `y` datetime/timedelta doesn't work because of line 170/172, where we do `y *= delta` or `y = y * delta`. `delta` is `stop-start`, so timedelta anyway, and if  `y` is also  datetime/timedelta, then the multiplication is not allowed.

Reply:
I have adapted the PR following @hameerabbasi 's `ndarray.view` idea. Now these lines are no longer present.